tests/tensorflow/test_printing.py

```diff
@@ -79,3 +79,67 @@ def test_unknown_shape():
     expected_out = 'Tensor(Placeholder):0,\tshape=Unknown\t"A:0"\n'
 
     assert std_out.getvalue() == expected_out.lstrip()
+
+
+@pytest.mark.usefixtures("run_with_tensorflow")
+@run_in_graph_mode
+def test_shape_index():
+    """Make sure graph indexing functions as expected."""
+
+    A = tf.compat.v1.placeholder('float', name='A',
+                                 shape=tf.TensorShape([None, None]))
+    x = tf.compat.v1.placeholder('float', name='x',
+                                 shape=tf.TensorShape([None, 1]))
+    y = tf.multiply(1.0, x, name='y')
+
+    z = tf.matmul(A, tf.add(y, y, name='x_p_y'), name='A_dot')
+
+
+    range_out = io.StringIO()
+    with redirect_stdout(range_out):        
+        tf_dprint(z, depth_lower=3, depth_upper=6)
+
+    expected_out = textwrap.dedent('''
+    ... Op(Add)\t"x_p_y"
+    ... |  Tensor(Mul):0,\tshape=[None, 1]\t"y:0"
+    ... |  |  Op(Mul)\t"y"
+    ... |  Tensor(Mul):0,\tshape=[None, 1]\t"y:0"
+    ... |  |  Op(Mul)\t"y"
+    ''')
+
+    assert range_out.getvalue() == expected_out.lstrip()
+
+    range_out = io.StringIO()
+    with redirect_stdout(range_out):        
+        tf_dprint(z)
+
+    expected_out = textwrap.dedent('''
+    Tensor(MatMul):0,\tshape=[None, 1]\t"A_dot:0"
+    |  Op(MatMul)\t"A_dot"
+    |  |  Tensor(Placeholder):0,\tshape=[None, None]\t"A:0"
+    |  |  Tensor(Add):0,\tshape=[None, 1]\t"x_p_y:0"
+    |  |  |  Op(Add)\t"x_p_y"
+    |  |  |  |  Tensor(Mul):0,\tshape=[None, 1]\t"y:0"
+    |  |  |  |  |  Op(Mul)\t"y"
+    |  |  |  |  |  |  Tensor(Const):0,\tshape=[]\t"y/x:0"
+    |  |  |  |  |  |  Tensor(Placeholder):0,\tshape=[None, 1]\t"x:0"
+    |  |  |  |  Tensor(Mul):0,\tshape=[None, 1]\t"y:0"
+    |  |  |  |  |  ...
+    ''')
+    assert range_out.getvalue() == expected_out.lstrip()
+
+@pytest.mark.usefixtures("run_with_tensorflow")
+@pytest.mark.xfail(strict=True)
+@run_in_graph_mode
+def test_shape_index_type():
+    """Make sure graph indexing functions as expected."""
+
+    A = tf.compat.v1.placeholder('float', name='A',
+                                 shape=tf.TensorShape([None, None]))
+    x = tf.compat.v1.placeholder('float', name='x',
+                                 shape=tf.TensorShape([None, 1]))
+    y = tf.multiply(1.0, x, name='y')
+
+    z = tf.matmul(A, tf.add(y, y, name='x_p_y'), name='A_dot')
+
+    tf_dprint(z, depth_lower=3.2, depth_upper=6.2)    
```

Comment:
This isn't a real test.

Reply:
Updated in my latest commit, let me know if it looks better.